### PR TITLE
Chore: Adding Open With network layer

### PR DESCRIPTION
**Pull request by jeremypress**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jeremypress has signed the CLA. Thanks for the pull request!

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #519

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

File: src/api/OpenWith.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+     *
+     * @param {string} [id] - a box file id
+     * @return {string} base url for files
+     */
+    getUrl(id: string): string {
+        if (!id) {
+            throw new Error('Missing file id!');
+        }
+
+        return `${this.getBaseApiUrl()}/files/${id}/open_with_integrations`;
```
Comment (priyajeet) [inline src/api/OpenWith.js]: Should this inherit from File.js?

File: src/api/Base.js [hunk 2]
```diff
@@ -341,9 +341,8 @@ class Base {
 
         // $FlowFixMe
         const xhrMethod: Function = this.xhr[method].bind(this.xhr);
-
         try {
-            const { data } = await xhrMethod({ id: getTypedFileId(id), url, ...requestData });
+            const { data } = await xhrMethod({ ...requestData, id: getTypedFileId(id), url });
```
Comment (priyajeet) [inline src/api/Base.js]: Not sure if this should be done. And likely will break stuff. Because now you are sending a fileId always and in some cases we want a folder id. Don't think any id should be sent here, and callers should be sending those

Comment (priyajeet) [reply src/api/Base.js]: Base.js should not have anything related to file or folder

Comment (jeremypress) [reply src/api/Base.js]: I'll move it back

The reason why we set the ID here is to help enforce that the ID is getting typed correctly. If folder related stuff wants to start using this then we will need to move this out.

File: src/api/OpenWith.js (hunk 1 above)
Comment (jeremypress) [reply src/api/OpenWith.js]: the getURL method in file assumes you're only passing in an ID, we would need to pass in a full suffix. We're not doing any other operations related to the file with this endpoint, so I think it's fine to leave as is.

File: src/api/AppIntegrations.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        if (!fileId) {
+            throw new Error('Missing file id!');
+        }
+
+        const executeURL = `${this.getUrl(integrationId)}/execute`;
+        const body = {
+            data: {
+                item: {
+                    id: fileId,
+                    type: 'file'
```
Comment (JustinHoldstock) [inline src/api/AppIntegrations.js]: Should we put `file` as a constant somewhere, and reuse it?

File: src/api/APIFactory.js [hunk 4]
```diff
@@ -206,6 +218,21 @@ class APIFactory {
             delete this.fileCollaboratorsAPI;
         }
 
+        if (this.feedItemsAPI) {
```
Comment (jstoffan) [inline src/api/APIFactory.js]: If this change isn't related to this patchset, can we move it to another PR? It'll help ensure that changes don't get inadvertently lost if we need to revert.

File: src/api/AppIntegrations.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+import { TYPE_FILE } from '../constants';
+
+class AppIntegrations extends Base {
+    /**
+     * API URL for Open With
+     *
+     * @param {string} [id] - a box integration app ID
+     * @return {string} base url for app integrations
+     */
+    getUrl(id: string): string {
```
Comment (jstoffan) [inline src/api/AppIntegrations.js]: Minor: Rename to `integrationId` to be consistent with the `execute` method.
